tests/Recurrence.test.ts

```diff
@@ -28,6 +28,78 @@ describe('Recurrence', () => {
             dueDate: null,
         });
     });
+
+    it('creates a recurrence the next month, even on the 31st', () => {
+        // Arrange
+        const recurrence = Recurrence.fromText({
+            recurrenceRuleText: 'every month',
+            startDate: null,
+            scheduledDate: null,
+            dueDate: moment('2022-01-31').startOf('day'),
+        });
+
+        // Act
+        const next = recurrence!.next();
+
+        // Assert
+        expect(next!.startDate).toBeNull();
+        expect(next!.scheduledDate).toBeNull();
+        expect(next!.dueDate!.isSame(moment('2022-02-28'))).toStrictEqual(true);
```

Comment:
I think these tests might be clearer to interpret if they were in the parameterised recurrence tests in `Test.task.ts`.

Later on, I might move those tests to here, although that's obviously outside the scope of this PR.